ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -117,11 +149,23 @@ def listener_callback(self, activity_msg):
             # Update state tracking vars
             self._previous_step = self._current_step
             self._current_step = self._task.state
+
         except transitions.core.MachineError as e:
             pass
 
         self.publish_task_state_message()
 
+        # Check to see if this new state has a timer associated with it
+        try:
+            if self._task.machine.states[self._task.state].timer_length > 0:
+                # Spawn a thread to track the timer and publish state messages
+                t = threading.Thread(target=self.task_timer_thread)
+                t.start()
+                self._timer_active = True
```

Comment:
Yeah, I wasn't sure how to approach this one really. I used the thread, rather than just doing the timer loop in the callback, because then we'd be able to receive new activity subscription callbacks and update the current activity while the timer loop is running, but there is probably a cleaner solution. All we need is to be able to send the task update messages in the timer loop while still responding to new activities.